### DISCUSSION TITLE
choose lsp implementation based upon file path

### DISCUSSION THREAD

**Discussion by mountainpath9** [helix-editor/helix#12240] [Q&A]
I have a repo with the following structure

```
deno.json
package.json
deno/src/**/*.ts                           # needs deno lsp
ts/src/**/*.ts                             # needs nodets lsp
```

and want to create a `.helix/languages.toml` file such that the appropriate lsp is started depending on the opened typescript file.

It seems like https://github.com/helix-editor/helix/pull/8696 was intended to address this, but it's not clear that it does. 

The workspace is structured with both `deno.json` and `package.json` in the root so that both `yarn` and `deno` can be executed from the root.

Should the [language server configuration](https://docs.helix-editor.com/languages.html#language-server-configuration)  support file globbing patterns in  a way similar to the [language configuration](https://docs.helix-editor.com/languages.html#file-type-detection-and-the-file-types-key) ?

**Comment by kirawi** [comment]
See https://github.com/helix-editor/helix/discussions/8372#discussioncomment-9716443

**Reply by mountainpath9** [reply]
Thanks though that discussion link is for a different problem. It covers selecting the lsp server based upon the presence of distinguishing files in the root of an lsp workspace. For a repo as I describe above - that's insufficient, and the lsp needs to be selected based upon source file path.
